tests/utils.go

```diff
@@ -2646,10 +2657,10 @@ func WaitForPodToDisappearWithTimeout(podName string, seconds int) {
 func WaitForVirtualMachineToDisappearWithTimeout(vmi *v1.VirtualMachineInstance, seconds int) {
 	virtClient, err := kubecli.GetKubevirtClient()
 	ExpectWithOffset(1, err).ToNot(HaveOccurred())
-	EventuallyWithOffset(1, func() bool {
-		_, err := virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &metav1.GetOptions{})
-		return errors.IsNotFound(err)
-	}, seconds, 1*time.Second).Should(BeTrue())
+	EventuallyWithOffset(1, func() error {
+		_, err := virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.GetName(), &metav1.GetOptions{})
+		return err
+	}, seconds, 1*time.Second).Should(And(HaveOccurred(), WithTransform(errors.IsNotFound, BeTrue())), "The VMI should be gone within the given timeout")
```

Comment:
ditto

Reply:
This one looks good to me.
Without this you need to have conditions inside the Eventually.

An alternative is to create our own custom matchers.

Reply:
here [this] is not the case: 
as @phoracek  suggested, I would like to assert on the `IsNotFound` only after I verified the error exists,
to avoid nil exception. I will change the `AND` to `SatisfyAll` which is equivalent but makes it more readble.
Will that be OK you think?

Reply:
`SatisfyAll` sounds good to me.